modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/twostep/GridReduceQueryExecutor.java

```diff
@@ -221,17 +223,19 @@ private void fail(ReduceQueryRun r, UUID nodeId, String msg, byte failCode, int
         if (r != null) {
             CacheException e;
 
-            if (failCode == GridQueryFailResponse.CANCELLED_BY_ORIGINATOR) {
-                e = new CacheException("Failed to execute map query on remote node [nodeId=" + nodeId +
-                    ", errMsg=" + msg + ']', new QueryCancelledException());
-            }
-            else if (failCode == GridQueryFailResponse.RETRY_QUERY) {
-                e = new CacheException("Failed to execute map query on remote node [nodeId=" + nodeId +
-                    ", errMsg=" + msg + ']', new QueryRetryException(msg));
+            String mapperFailedMsg = "Failed to execute map query on remote node [nodeId=" + nodeId +
+                ", errMsg=" + msg + ']';
+
+            if (failCode == GridQueryFailResponse.CANCELLED_BY_ORIGINATOR)
+                e = new CacheException(mapperFailedMsg, new QueryCancelledException());
+            else if (failCode == GridQueryFailResponse.RETRY_QUERY)
+                e = new CacheException(mapperFailedMsg, new QueryRetryException(msg));
+            else if (sqlErrCode == IgniteQueryErrorCode.QUERY_OUT_OF_MEMORY) {
```

Comment:
Seems that this is logical failure:
 - we DO WANT track map OOM errors as sql.queries.system failed queries counter, but 
 - we DO NOT WANT to suppress sql error code if OOM happened on map phase (e.g. OOM should be propagated to reduce phase and finally to user)

Reply:
Discussed off-line.